src/box/sql/mem.c

```diff
@@ -988,6 +1023,69 @@ double_to_uint_forced(struct Mem *mem)
 	return res;
 }
 
+static inline int
+double_to_dec(struct Mem *mem)
+{
+	assert(mem->type == MEM_TYPE_DOUBLE);
+	double d = mem->u.r;
+	if (d >= 1e38 || d <= -1e38)
```

Comment:
Why do you need this additional check?

Reply:
Because cast of values more than 1e38 and less than -1e38 are out of DECIMAL range. So CAST should fail.

Reply:
Or do you mean that we just have to see result of `decimal_from_double()`?

Reply:
> Or do you mean that we just have to see result of decimal_from_double()?

Yep

Reply:
Fixed.